readthedocs/core/views/serve.py

```diff
@@ -269,3 +272,74 @@ def robots_txt(request, project):
         return HttpResponse(open(fullpath).read(), content_type='text/plain')
 
     return HttpResponse('User-agent: *\nAllow: /\n', content_type='text/plain')
+
+
+@map_project_slug
+# TODO: make this cache dependent on the project's slug
+@cache_page(60 * 60 * 24 * 3)  # 3 days
+def sitemap_xml(request, project):
+
+    def priorities_generator():
+        """
+        Generator returning ``priority`` needed by sitemap.xml.
+
+        It generates values from 1 to 0.1 by decreasing in 0.1 on each
+        iteration. After 0.1 is reached, it will keep returning 0.1.
+        """
+        priorities = [1, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2]
+        for p in priorities:
+            yield p
+
+        while True:
+            yield 0.1
+
+    def changefreqs_generator():
+        """
+        Generator returning ``changefreq`` needed by sitemap.xml.
+
+        It returns ``daily`` on first iteration, then ``weekly`` and then it
+        will return always ``monthly``.
+        """
+        changefreqs = ['daily', 'weekly']
+        for c in changefreqs:
+            yield c
+
+        while True:
+            yield 'monthly'
```

Comment:
I'm using `monthly` because I think `never` is too aggressive. If the tag is removed and a branch is created with the same name, we will want bots to revisit this.

> NOTE: maybe this should be a comment in the code itself.